src/utils/daos_autotest.c

```diff
@@ -479,7 +541,7 @@ kv_insert128(void)
 	int		rc;
 
 	new_oid();
-	daos_obj_generate_oid(coh, &oid, DAOS_OF_KV_FLAT, OC_SX, 0, 0);
+	daos_obj_generate_oid(coh, &oid, DAOS_OF_KV_FLAT, 0, 0, 0);
```

Comment:
???

Reply:
That was my interpretation of how to use the API to auto-select the object class.
This file <> states:
" * \param[in]   cid     Class Identifier. This setting is for advanced users who
 *                      are knowledgeable on the specific oclass being set and
 *                      what that means for the object in the current system and
 *                      the container it's in.
 *                      Setting this to 0 (unknown) will check if there are any
 *                      hints specified and use an oclass accordingly. If there
 *                      are no hints specified we use the container properties
 *                      to select the object class."

I read that as setting cid and hints to 0 it will auto select object class from the container properties itself. This was attempted to be used to fulfill the requirement of "use the new API to auto-select the object class".

If my interpretation is incorrect can you let me know where to find this portion of the code?

Reply:
Discussed and this is fine.